src/test/java/com/exasol/bucketfs/client/DownloadCommandIT.java

```diff
@@ -0,0 +1,162 @@
+package com.exasol.bucketfs.client;
+
+import static com.exasol.bucketfs.client.IntegrationTestSetup.content;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.itsallcode.junit.sysextensions.AssertExit.assertExitWithStatus;
+import static picocli.CommandLine.ExitCode.OK;
+import static picocli.CommandLine.ExitCode.SOFTWARE;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.itsallcode.io.Capturable;
+import org.itsallcode.junit.sysextensions.ExitGuard;
+import org.itsallcode.junit.sysextensions.SystemErrGuard;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+import com.exasol.bucketfs.BucketAccessException;
+
+@ExtendWith(ExitGuard.class)
+@ExtendWith(SystemErrGuard.class)
+class DownloadCommandIT {
+
+    private static final IntegrationTestSetup SETUP = new IntegrationTestSetup();
+
+    @BeforeAll
+    static void beforeAll() throws BucketAccessException, InterruptedException {
+        // parent needs to be created *after* child, otherwise HTTP client method send() blocks.
+        SETUP.createRemoteFiles("a.txt", "folder/aa.txt", "problem/ambigue/any.txt", "problem/ambigue");
+    }
+
+    @AfterAll
+    static void after() {
+        SETUP.stop();
+    }
+
+    // [itest->dsn~copy-command-copies-file-to-bucket~1]
+    @Test
+    void testFailureDownloadWithMalformedBucketFsUrl(final Capturable stream) {
+        final BFSC client = BFSC.create("cp", "bfs://illegal/", "some_file");
+        stream.capture();
+        assertExitWithStatus(SOFTWARE, () -> client.run());
+        assertThat(stream.getCapturedData(), startsWith("E-BFSC-5: Invalid BucketFS URL: 'bfs://illegal/'"));
+    }
+
+    // [itest->dsn~copy-command-copies-file-from-bucket~1]
+    @Test
+    void testFailureDownloadNonExistingObject(final Capturable stream) {
+        verifyNoSuchFile("non-existing-object", stream);
+    }
+
+    @Test
+    void testFailureDownloadRoot(final Capturable stream) {
+        verifyNoSuchFile("", stream);
+    }
+
+    private void verifyNoSuchFile(final String remote, final Capturable stream) {
+        final BFSC client = BFSC.create("cp", bfsUri(remote), "some_file");
+        stream.capture();
+        assertExitWithStatus(SOFTWARE, () -> client.run());
+        assertThat(stream.getCapturedData().trim(),
+                matchesRegex("E-BFSC-10: Cannot download '.*/" + remote + "': No such file or directory."));
+    }
+
+    private String bfsUri(final String pathInBucket) {
+        return SETUP.getDefaultBucketUriToFile(pathInBucket);
+    }
+
+    @Test
+    void testFailureDownloadDirectoryWithoutRecursiveFlag(@TempDir final Path tempDir, final Capturable stream) {
```

Comment:
FYI: You can replace parameter `@TempDir final Path tempDir` with a field, so you don't need the parameter. Depends on what you prefer, both is fine for me.